copy.go

```diff
@@ -21,19 +21,50 @@ import (
 	"fmt"
 
 	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"golang.org/x/sync/semaphore"
 	"oras.land/oras-go/v2/content"
 	"oras.land/oras-go/v2/errdef"
 	"oras.land/oras-go/v2/internal/cas"
 	"oras.land/oras-go/v2/internal/graph"
 	"oras.land/oras-go/v2/internal/status"
 )
 
+// defaultConcurrency is the default concurrency limit.
+// The value is consistent with dockerd and containerd.
+const defaultConcurrency = int64(3)
+
+// CopyOptions contains parameters for oras.Copy.
+type CopyOptions struct {
+	CopyGraphOptions
+	// RootFilter filters the descriptor that is to be used as the root node for
+	// copy.
+	RootFilter func(ctx context.Context, root ocispec.Descriptor) (desc ocispec.Descriptor, found bool, err error)
```

Comment:
Can you help explain the scenarios we need to use this root filter?

Reply:
Some scenarios I can think of are:
1. Filter manifests of allowed media types for copy. e.g. Clients may want to copy only when the resolved manifest is a OCI manifest instead of a Docker manifest, like https://github.com/Wwwsylvia/oras-go/blob/copy_options/copy_test.go#L430-L439.
2. Filter manifests of allowed platforms for copy. e.g. Clients may want to pick a manifest of a specific platform from a resolved manifest list, like https://github.com/Wwwsylvia/oras-go/blob/copy_options/copy_test.go#L468-L489.

When there are multiple matching manifests, clients can decide which manifest to return. Or they can generate a new manifest list and return the descriptor of the manifest list. Maybe we can add some utility functions in the future to support filtering and generating manifest list.

Does it make sense?

Reply:
Sorted out offline with Sylvia. The second use case is the most common one(like `--platform` flag for crane and ctr image pull) and the current RootFilter implementation does cover it.